homeassistant/components/homeassistant/triggers/homeassistant.py

```diff
@@ -30,24 +30,17 @@ async def async_attach_trigger(
     job = HassJob(action, f"homeassistant trigger {trigger_info}")
 
     if event == EVENT_SHUTDOWN:
-
-        @callback
-        def hass_shutdown(event):
-            """Execute when Home Assistant is shutting down."""
-            hass.async_run_hass_job(
-                job,
-                {
-                    "trigger": {
-                        **trigger_data,
-                        "platform": "homeassistant",
-                        "event": event,
-                        "description": "Home Assistant stopping",
-                    }
-                },
-                event.context,
-            )
-
-        return hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, hass_shutdown)
+        return hass.async_add_shutdown_job(
+            job,
+            {
+                "trigger": {
+                    **trigger_data,
+                    "platform": "homeassistant",
+                    "event": event,
```

Comment:
I'm not sure this is valid, does it have to be an event object?

Reply:
Trigger data is documented here: https://www.home-assistant.io/docs/automation/templating#available-trigger-data, but the trigger data for the `homeassistant` trigger is undocumented. I still think this should be considered a breaking change though, in case someone relies on it.

Reply:
Marking this resolved as its labeled a breaking change now